### PR TITLE
[UXE-4029] feat: redirect to billing page when client status is BLOCKED or DEFAULTING

### PR DESCRIPTION
**Pull request by HerbertJulio**
### What is the new behavior introduced by this PR?
Redirect to billing page when client status is BLOCKED or DEFAULTING

### Does this PR introduce breaking changes?
- [x] No
- [ ] Yes 

### Does this PR introduce UI changes? Add a video or screenshots here.
NO

### Does it have a link on Figma?
NO

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] TYPE: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Review by matheus-sunderhus — APPROVED** [review]

File: src/router/hooks/beforeEachRoute.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
 import { useLoadingStore } from '@/stores/loading'
+import { billingRoutes } from '@/router/routes/billing-routes'
 
 /** @type {import('vue-router').NavigationGuardWithThis} */
 export default async function beforeEachRoute(to, __, next) {
   const accountStore = useAccountStore()
   const loadingStore = useLoadingStore()
   const helpCenterStore = useHelpCenterStore()
+  const isPrivateRoute = !to.meta.isPublic
+  const userNotIsLoggedIn = !accountStore.hasActiveUserId
```
Comment (william-tome) [inline src/router/hooks/beforeEachRoute.js:15]: nitpick 1/3

```suggestion
  const isLoggedIn = accountStore.hasActiveUserId
```

File: src/router/hooks/beforeEachRoute.js [hunk 2]
```diff
@@ -32,16 +36,14 @@ export default async function beforeEachRoute(to, __, next) {
     loadingStore.finishLoading()
   }
 
-  if (!accountStore.hasActiveUserId && !to.meta.isPublic) {
+  if (userNotIsLoggedIn && isPrivateRoute) {
```
Comment (william-tome) [inline src/router/hooks/beforeEachRoute.js:39]: nitpick 2/3
```suggestion
  if (!isLoggedIn && isPrivateRoute) {
```

File: src/router/hooks/beforeEachRoute.js [hunk 3]
```diff
@@ -53,12 +55,17 @@ export default async function beforeEachRoute(to, __, next) {
...
 
       accountStore.setAccountData(accountInfo)
-
-      return next()
     } catch {
       return next('/login')
     }
   }
 
+  if (accountStore.hasActiveUserId && isPrivateRoute && shouldRedirectToBilling) {
```
Comment (william-tome) [inline src/router/hooks/beforeEachRoute.js:63]: nitpick 3/3
```suggestion
  if (isLoggedIn && isPrivateRoute && shouldRedirectToBilling) {
```

**Review by william-tome — APPROVED** [review]
Pequenos detalhes. Sugestões não bloqueantes.

**Pull request merged**